tsdb/engine/tsm1/wal.go

```diff
@@ -380,19 +387,19 @@ func (w *WriteWALEntry) Encode(dst []byte) ([]byte, error) {
 		}
 		n++
 
-		// Make sure we have enough space in our buf before copying.  If not,
-		// grow the buf.
-		if len(k)+2+len(v)*8+4 > len(dst)-n {
-			grow := make([]byte, len(dst)*2)
-			dst = append(dst, grow...)
-		}
-
 		n += copy(dst[n:], u16tob(uint16(len(k))))
 		n += copy(dst[n:], []byte(k))
 
 		n += copy(dst[n:], u32tob(uint32(len(v))))
 
 		for _, vv := range v {
+
+			// Grow our slice if needed
+			if len(dst[:n])+16 > len(dst) {
```

Comment:
What is this change for? Didn't you make the buffer large enough up above? So many `v` s. :-)